File: szxy-funplay-parent/funplay-user-web/src/main/java/com/funplay/user/controller/UserController.java
package com.funplay.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.funplay.pojo.User;
import com.funplay.user.service.UserService;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import util.PhoneFormatCheckUtils;

@RestController
@RequestMapping("/user")
public class UserController {
    @Reference
    private UserService userService;

    @RequestMapping("/add")
    public Result add(@RequestBody User user, String sms){
       if(!userService.checkSmsCode(user.getPhone(),sms)) {
           return new Result(false, "验证码输入有误");
       }
       try{
           userService.add(user);
           return new Result(true,"注册成功");
       }catch (Exception e){
           e.printStackTrace();
           return  new Result(false,"注册失败");
       }
    }

    @RequestMapping("/sendCode")
    public Result sendCode(String phone){
        if(!PhoneFormatCheckUtils.isPhoneLegal(phone)){
            return  new Result(false,"手机格式不正确");
        }
        try{
            userService.createSmsCode(phone);
            return new Result(true,"验证码发送成功");
        }catch(Exception e){
            e.printStackTrace();
            return new Result(false,"验证码发送失败");
        }
    }
}

File: szxy-funplay-parent/funplay-user-web/src/main/webapp/js/controller/userController.js
app.controller('userController',function ($scope,$location,$controller,userService) {
    $scope.entity={};

                //注册
    $scope.reg=function(){
        if($scope.entity.username==null||$scope.entity.username==''){
            alert("请输入用户名");
            return;
        }
        if($scope.entity.password==null||$scope.entity.password==''){
            alert("请输入密码");
            return;
        }
        if($scope.entity.phone==null||$scope.entity.phone==''){
            alert("请输入手机号码");
            return;
        }
        if($scope.sms==null||$scope.sms==''){
            alert("请输入验证码");
            return;
        }
        if($scope.entity.password!=$scope.password){
            alert("两次密码不一致请重新输入");
            $scope.entity.password='';
            $scope.password='';
            return;//俩次密码不一致
        }
       userService.add($scope.entity,$scope.sms).success(
           function (response) {
               if(response.success){
                   alert(response.message);
                   $scope.entity={};
                   $scope.password='';
                   $scope.sms='';
               }else{
                   alert(response.message);
               }

           }
       )
    }
    //发送验证码
    $scope.sendCode=function () {
        if($scope.entity.phone==null||$scope.entity.phone==''){
            alert("请输入手机号");
            return;
        }
        userService.sendCode($scope.entity.phone).success(
            function (response) {
                alert(response.message);
            }
        )
    }
});